src/jira/client/jira-client-issue-key-helper.ts

```diff
@@ -0,0 +1,172 @@
+import { createHashWithSharedSecret } from "utils/encryption";
+import {
+	JiraAssociation,
+	JiraCommit,
+	JiraRemoteLink
+} from "interfaces/jira";
+
+export interface IssueKeyObject {
+	issueKeys?: string[];
+	associations?: JiraAssociation[];
+}
+
+// Max number of issue keys we can pass to the Jira API
+export const ISSUE_KEY_API_LIMIT = 500;
+
+/**
+ * Truncates to ISSUE_KEY_API_LIMIT elements in an array
+ */
+export const truncate = (array: unknown[]) => {
+	return array.slice(0, ISSUE_KEY_API_LIMIT);
+};
+
+/**
+ * Truncates branches, commits and PRs to their first ISSUE_KEY_API_LIMIT issue keys
+ */
+export const truncateIssueKeys = (repositoryObj) => {
+	updateRepositoryIssueKeys(repositoryObj, truncate);
+};
+
+/**
+ * Get truncated issue keys and associations based on the ISSUE_KEY_API_LIMIT.
+ */
+export const getTruncatedIssueKeys = (data: IssueKeyObject[] = []): IssueKeyObject[] =>
+	data.map((value: IssueKeyObject) => {
+		const truncatedValue: IssueKeyObject = {};
+
+		if (value?.issueKeys) {
+			truncatedValue.issueKeys = value.issueKeys.slice(0, ISSUE_KEY_API_LIMIT);
+		}
+
+		const association = findIssueKeyAssociation(value);
+		if (association?.values) {
+			truncatedValue.associations = [
+				{
+					associationType: association.associationType,
+					values: association.values.slice(0, ISSUE_KEY_API_LIMIT)
+				}
+			];
+		}
+
+		return truncatedValue;
+	});
+
+/**
+ * Returns if the max length of the issue key field is within the limit
+ */
+export const withinIssueKeyLimit = (resources: IssueKeyObject[]): boolean => {
+	if (!resources) return true;
+	const issueKeyCounts = resources.map((r) => r.issueKeys?.length || findIssueKeyAssociation(r)?.values?.length || 0);
+	return Math.max(...issueKeyCounts) <= ISSUE_KEY_API_LIMIT;
+};
+
+/**
+ * Deduplicates issueKeys field for branches and commits
+ */
+export const dedupIssueKeys = (repositoryObj) => {
+	updateRepositoryIssueKeys(repositoryObj, uniq);
+};
+
+/**
+ * Runs a mutating function on all branches, commits and PRs
+ * with issue keys in a Jira Repository object
+ */
+export const updateRepositoryIssueKeys = (repositoryObj, mutatingFunc) => {
+	if (repositoryObj.commits) {
+		repositoryObj.commits = updateIssueKeysFor(repositoryObj.commits, mutatingFunc);
+	}
+
+	if (repositoryObj.branches) {
+		repositoryObj.branches = updateIssueKeysFor(repositoryObj.branches, mutatingFunc);
+		repositoryObj.branches.forEach((branch) => {
+			if (branch.lastCommit) {
+				branch.lastCommit = updateIssueKeysFor([branch.lastCommit], mutatingFunc)[0];
+			}
+		});
+	}
+
+	if (repositoryObj.pullRequests) {
+		repositoryObj.pullRequests = updateIssueKeysFor(repositoryObj.pullRequests, mutatingFunc);
+	}
+};
+
+/**
+ * Finds the first association of type "issueIdOrKeys" in a given resource.
+ */
+export const findIssueKeyAssociation = (resource: IssueKeyObject): JiraAssociation | undefined => {
+	return resource.associations?.find(a => a.associationType == "issueIdOrKeys");
```

Comment:
nit `===`

Reply:
This is copied from jiraclients file and landed here to tidy up. no new logic. :excite: